provisioning/miniprov/hare_mp/main.py

```diff
@@ -225,6 +228,104 @@ def init(args):
         raise RuntimeError(f'Error while initializing cluster :key={error}')
 
 
+def __is_systemd_enabled() -> bool:
+    try:
+        cmd = ['systemctl']
+        execute(cmd)
+    except Exception:
+        return False
+    return True
+
+
+def start_hax_with_systemd():
+    cmd = ['systemctl', 'start', 'hare-hax']
+    execute(cmd)
+
+
+# def start_hax_without_systemd():
+#     try:
+#         path = os.getenv('PATH', '')
+#         path += os.pathsep + '/opt/seagate/cortx/hare/bin'
+#         path += os.pathsep + '/opt/seagate/cortx/hare/libexec'
+#         python_path = os.pathsep.join(sys.path)
+#         hare_log_file = '/var/log/seagate/hare/cortx-hare.log'
+#         var_hax_path = '/var/motr/hax'
+#         if not os.path.exists(var_hax_path):
+#             os.makedirs(var_hax_path)
+#         logger = logging.getLogger('hax-log')
+#         fhandler = logging.handlers.RotatingFileHandler(hare_log_file,
+#                                                         maxBytes=LOG_FILE_SIZE,
+#                                                         mode='a',
+#                                                         backupCount=5,
+#                                                         encoding=None,
+#                                                         delay=False)
+#         logger.addHandler(fhandler)
+#         cmd = ['hax']
+#         execute_no_communicate(cmd, env={'PYTHONPATH': python_path,
+#                                          'PATH': path,
+#                                          'LC_ALL': "en_US.utf-8",
+#                                          'LANG': "en_US.utf-8"},
+#                                working_dir=var_hax_path,
+#                                out_file=LogWriter(logger, fhandler))
+#     except Exception as error:
+#         raise RuntimeError(f'Error while initializing cluster :key={error}')
+
+def _start_consul(utils: Utils, stop_event: Event, hare_local_dir: str):
+    log_dir = f'{hare_local_dir}/consul/log'
+    if not os.path.exists(log_dir):
+        os.makedirs(log_dir)
+    data_dir = f'{hare_local_dir}/consul/data'
+    if not os.path.exists(data_dir):
+        os.makedirs(data_dir)
+    config_dir = f'{hare_local_dir}/consul/config'
+    if not os.path.exists(config_dir):
+        os.makedirs(config_dir)
+    peers = ['consul-server']
```

Comment:
@mssawant  I guess this will be used in 'retry-join'. Are we not going to use endpoint provided by 'cortx>external>consul>endpoints' here.

Reply:
Yes, once we have merged all the new conf-store key changes, we can read the consul endpoint from conf-store. This can be done separately.